test/select.spec.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+/* eslint-env mocha */
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+const { expect } = chai
+const MSS = require('../')
+const pipe = require('it-pipe')
+const { collect } = require('streaming-iterables')
+const Crypto = require('crypto')
+const BufferList = require('bl')
+const Pair = require('it-pair')
+const throwsAsync = require('./helpers/throws-async')
+
+describe('select', () => {
+  it('should select from single protocol', async () => {
+    const protocol = '/echo/1.0.0'
+    const muxedStream = Pair()
```

Comment:
Just a nit: `muxedStream` is a bit of a misnomer as this is an unmuxed stream we're doing protocol selection on, there's no multiplexer involved. No need to change it, but it could be confusing.